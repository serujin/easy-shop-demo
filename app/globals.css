@import url("https://p.typekit.net/p.css?s=1&k=gws6bus&ht=tk&f=139&a=10612664&app=typekit&e=css");
@font-face {
  font-family: "proxima-nova";
  src: url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/l?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("woff2"),
    url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/d?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("woff"),
    url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/a?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("opentype");
  font-display: auto;
  font-style: normal;
  font-weight: 700;
  font-stretch: normal;
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  font-family: "proxima-nova";
}

html {
  color: #404040;
  font-size: 14px;
  background-color: #efefef;
}

body {
  display: flex;
  flex-direction: column;
}

header {
  display: flex;
  background-color: #4e2817;
  justify-content: center;
}

a {
  text-decoration: none;
  color: black;
}
