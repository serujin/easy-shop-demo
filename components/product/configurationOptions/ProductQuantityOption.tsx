import React, { ChangeEvent } from "react";
import styles from "./ProductQuantityOption.module.css";

export type ProductQuantityOptionType = {
  discount?: number;
  price?: Amount;
  value: string;
  isLast?: boolean;
};

type ProductQuantityOptionProps = ProductQuantityOptionType & {
  handleQuantityChange: (event: ChangeEvent<HTMLInputElement>) => void;
  optionNumber: number;
  selectedQuantity: string;
};

const createId = (optionNumber: number, value: string) =>
  `quantity-${value}-${optionNumber}`;

export default function ProductQuantityOption(
  props: ProductQuantityOptionProps
) {
  const {
    discount,
    handleQuantityChange,
    optionNumber,
    price,
    selectedQuantity,
    value,
    isLast = false,
  } = props;

  const id = createId(optionNumber, value);
  const discountLabel = discount ? `Save ${discount}%` : "";
  const priceLabelText = price ? `${price.quantity}${price.currency}` : "";

  return (
    <tr className={styles.quantityRow}>
      <td className={styles.inputCell}>
        <input
          type="radio"
          id={id}
          name="quantities"
          value={value}
          checked={selectedQuantity === value}
          onChange={handleQuantityChange}
        />
      </td>
      {isLast && (
        <td colSpan={3}>
          <label htmlFor={id}>{value}</label>
        </td>
      )}
      {!isLast && (
        <td>
          <label htmlFor={id}>{value}</label>
        </td>
      )}
      {price && (
        <td className={styles.rightAlign}>
          <label htmlFor={id}>{priceLabelText}</label>
        </td>
      )}
      {discount && (
        <td className={styles.rightAlign}>
          <p>{discountLabel}</p>
        </td>
      )}
    </tr>
  );
}
