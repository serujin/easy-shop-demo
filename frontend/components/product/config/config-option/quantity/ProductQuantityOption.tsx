import React, { ChangeEvent } from 'react'
import styles from './ProductQuantityOption.module.css'

export type ProductQuantityOptionType = {
  discount?: number
  price?: Amount
  value: string
  isLast?: boolean
}

type ProductQuantityOptionProps = ProductQuantityOptionType & {
  handleQuantityChange: (event: ChangeEvent<HTMLInputElement>) => void
  optionNumber: number
  selectedQuantity: string
}

const getId = (optionNumber: number, value: string) => {
  const valueId = value.toLowerCase().replace(' ', '-')
  return `quantity-${valueId}-${optionNumber}`
}

const getDiscountLabel = (discount?: number) =>
  discount ? `Save ${discount}%` : ''

const getPriceLabel = (price?: Amount) =>
  price ? `${price.currency}${price.quantity}` : ''

const getValueTableCell = (isLast: boolean, id: string, value: string) =>
  isLast ? (
    <td colSpan={3}>
      <label htmlFor={id}>{value}</label>
    </td>
  ) : (
    <td>
      <label htmlFor={id}>{value}</label>
    </td>
  )

export default function ProductQuantityOption(
  props: ProductQuantityOptionProps,
) {
  const {
    discount,
    handleQuantityChange,
    optionNumber,
    price,
    selectedQuantity,
    value,
    isLast = false,
  } = props

  const id = getId(optionNumber, value)
  const discountLabel = getDiscountLabel(discount)
  const priceLabelText = getPriceLabel(price)
  const discountTableCellClassName = `${styles.rightAlign} ${styles.discount}`

  return (
    <tr className={styles.quantityRow}>
      <td className={styles.inputCell}>
        <input
          type="radio"
          id={id}
          name="quantities"
          value={value}
          checked={selectedQuantity === value}
          onChange={handleQuantityChange}
        />
      </td>
      {getValueTableCell(isLast, id, value)}
      {price && (
        <td className={styles.rightAlign}>
          <label htmlFor={id}>{priceLabelText}</label>
        </td>
      )}
      {discount && (
        <td className={discountTableCellClassName}>
          <label htmlFor={id}>{discountLabel}</label>
        </td>
      )}
    </tr>
  )
}
