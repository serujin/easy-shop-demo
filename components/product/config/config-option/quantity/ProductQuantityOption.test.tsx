import React from 'react'
import { render } from '@testing-library/react'
import ProductQuantityOption from './ProductQuantityOption'
import { describe, expect, jest, it } from '@jest/globals'

describe('ProductQuantityOption component', () => {
  describe('with only required props', () => {
    it('should render correctly', () => {
      const { container } = render(
        <table>
          <tbody>
            <ProductQuantityOption
              handleQuantityChange={jest.fn()}
              optionNumber={0}
              selectedQuantity="selectedQuantity"
              value="value"
            />
          </tbody>
        </table>,
      )

      expect(container).toMatchSnapshot()
    })
  })
  describe('with all props', () => {
    it('should render correctly', () => {
      const { container } = render(
        <table>
          <tbody>
            <ProductQuantityOption
              handleQuantityChange={jest.fn()}
              optionNumber={0}
              selectedQuantity="selectedQuantity"
              value="value"
              discount={10}
              price={{ quantity: 0, currency: '$' }}
              isLast
            />
          </tbody>
        </table>,
      )
      expect(container).toMatchSnapshot()
    })
  })
})
