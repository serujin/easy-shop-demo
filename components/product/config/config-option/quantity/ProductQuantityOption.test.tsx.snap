// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductQuantityOption component with all props should render correctly 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="quantityRow"
      >
        <td
          class="inputCell"
        >
          <input
            id="quantity-value-0"
            name="quantities"
            type="radio"
            value="value"
          />
        </td>
        <td
          colspan="3"
        >
          <label
            for="quantity-value-0"
          >
            value
          </label>
        </td>
        <td
          class="rightAlign"
        >
          <label
            for="quantity-value-0"
          >
            $0
          </label>
        </td>
        <td
          class="rightAlign discount"
        >
          <label
            for="quantity-value-0"
          >
            Save 10%
          </label>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`ProductQuantityOption component with only required props should render correctly 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="quantityRow"
      >
        <td
          class="inputCell"
        >
          <input
            id="quantity-value-0"
            name="quantities"
            type="radio"
            value="value"
          />
        </td>
        <td>
          <label
            for="quantity-value-0"
          >
            value
          </label>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
